### PR TITLE
Fix: Fixed an issue where shortcuts couldn't be created for commands

### LINKED ISSUE #15928: Bug: Can't put commands in shortcuts files
### Description

In Windows, u can create command in a shortcuts and it worked well on file explorer however in Files u cant create a shortcut with a command in it (It only worked on the desktop).

### Steps To Reproduce

1. Right click in a random folder (e.g Music folder)
2. Click New and then Shortcut 
3. Try to put a command (e.g winget upgrade --all) and it would say "Invalid location"

### Requirements

Files would not allow to put commands in shortcut.

### Files Version

3.5.0.0

### Windows Version

22631.3958

### Log File

[debug.log](https://github.com/user-attachments/files/16478384/debug.log)

### PR DESCRIPTION
**Pull request by Qin-shihuang**
**Resolved / Related Issues**

- Closes #15928

**Steps used to test these changes**

1. Right click in a folder
2. Click New -> Shortcut in the context menu
3. Type "cmd /?" in the textbox
4. Before: You got a prompt saying "Invalid location"
	After: Shortcut successfully created
5. (After) Open the shortcut and see the cmd help text

**-- Update --**
These modifications make the behavior of creating shortcuts more consistent with File Explorer.

These modifications can be tested by creating the following shortcuts, all of which are supported by File Explorer but not supported by Files before:
1. `cmd`
2. `cmd.exe`
3. `cmd /?`
4. `  cmd`
5. `"C:\Windows\System32\cmd.exe" /?`
6. `C:\Windows\System32\cmd.exe /?`
7. `"C:\Program Files\PowerShell\7\pwsh.exe" -c cmd /?`
8. `"https://www.google.com"`

Also fixed the issue 4 mentioned below, which is caused by Path.Exists considering Files' startup path as relative path.

### CODE REVIEW

**Comment by Qin-shihuang** [comment]
While working on this, I discovered several other issues:

1. This page cannot create shortcuts for drives
image

2. The default naming rule looks strange
image

File: src/Files.App/ViewModels/Dialogs/CreateShortcutDialogViewModel.cs [hunk 1]
```diff
@@ -45,20 +46,35 @@ public string DestinationItemPath
...
+					IsLocationValid = uri.IsWellFormedOriginalString();
+					_isLocationInPath = false;
+					_fullPath = DestinationItemPath;
 				}
 				catch (Exception)
 				{
+					DestinationPathExists = false;
 					IsLocationValid = false;
+					_isLocationInPath = false;
 				}
```
Comment (hishitetsu) [inline src/Files.App/ViewModels/Dialogs/CreateShortcutDialogViewModel.cs:78]: Is it unnecessary to set `_fullPath`?

File: src/Files.App/ViewModels/Dialogs/CreateShortcutDialogViewModel.cs [hunk 2]
```diff
@@ -75,6 +91,9 @@ public bool IsLocationValid
 			}
 		}
 
+		private bool _isLocationInPath;
```
Comment (hishitetsu) [inline src/Files.App/ViewModels/Dialogs/CreateShortcutDialogViewModel.cs:94]: Is the name of this field appropriate? I think a name like `_hasArguments` would be more appropriate.

File: src/Files.App/ViewModels/Dialogs/CreateShortcutDialogViewModel.cs [hunk 3]
```diff
@@ -115,12 +134,17 @@ private async Task CreateShortcutAsync()
...
-				destinationName = Path.GetFileName(DestinationItemPath);
-				if (string.IsNullOrEmpty(destinationName))
+				destinationName = Path.GetFileName(_fullPath);
+				if (_isLocationInPath)
+					arguments = DestinationItemPath.Split(' ')[1..].Aggregate(arguments, (current, arg) => current + arg + " ");
+
+				if(string.IsNullOrEmpty(destinationName))
 				{
+					
 					var destinationPath = DestinationItemPath.Replace('/', '\\');
```
Comment (hishitetsu) [inline src/Files.App/ViewModels/Dialogs/CreateShortcutDialogViewModel.cs:148]: Shouldn't `DestinationItemPath` be replaced with `_fullPath`?
Same for L158.

File: src/Files.Shared/Helpers/PathHelpers.cs [hunk 4]
```diff
@@ -63,5 +64,39 @@ public static bool IsSpecialFolder(string path)
...
+				{
+					UseShellExecute = false,
+					CreateNoWindow = true,
+					FileName = "where.exe",
+					Arguments = exeName,
+					RedirectStandardOutput = true
+				};
+				p.Start();
+				var output = p.StandardOutput.ReadToEnd();
+				p.WaitForExit();
```
Comment (hishitetsu) [inline src/Files.Shared/Helpers/PathHelpers.cs:83]: A timeout period should be set to avoid stoppages.

**Comment by Qin-shihuang** [comment]
Sorry for the low-quality code. I am rewriting a better version that is more consistent with Files Explorer's behavior.

**Comment by hishitetsu** [comment]
Don't mind! Your contribution is very helpful. It's working and I don't consider it low-quality.

**Review by hishitetsu — COMMENTED** [review]
This would cause PathHelpers.TryGetFullPath() to be called for every input, even when editing arguments.
It is better to call it only when the file name is changed.

**Review by hishitetsu — APPROVED** [review]
Nice work! LGTM
But as you say, there are many other things that need to be fixed in this feature.

**Pull request merged**